tests/pagination/src/commonTest/kotlin/OffsetBasedWithPagePaginationTest.kt

```diff
@@ -233,13 +233,11 @@ class OffsetBasedWithPagePaginationTest {
 }
 
 internal fun assertChainedCachesAreEqual(apolloStore: ApolloStore) {
-  val hasNextCache = atomic(false)
-  apolloStore.accessCache { cache ->
-    // First cache is always OptimisticCache
-    hasNextCache.value = cache.nextCache!!.nextCache != null
+  val dump = apolloStore.dump().filterKeys {
+    // Ignore optimistic cache for comparison
+    it != OptimisticNormalizedCache::class && it != OptimisticNormalizedCacheWrapper::class
```

Comment:
What is the reason those might be diferent?

Reply:
`MemoryCache` uses `OptimisticNormalizedCache::class` as a key in its `dump()` fun. Maybe not the most intuitive thing?

Reply:
Gotcha 👍  Yea, not the best API. Add it to the list of APIs to bikeshed?